### DISCUSSION TITLE
Does anyone understand why this alignment tag comes in late in my delta structure?  Is it a bug?

### DISCUSSION THREAD

**Discussion by mark8044** [singerdmx/flutter-quill#448] [General]
I have the following showing in my quill editor
```
Test 1
                                         Test 2
Test 3
```

Test 2 is centered using the `alignment: center` toolbar button

But my delta when I output as a string looks like this:

```
flutter: insert⟨ Test 1⏎Test 2 ⟩
flutter: insert⟨ ⏎ ⟩ + {align: center}
flutter: insert⟨ Test 3⏎⏎ ⟩
```

It looks like the alignment center is put in after the carriage return after Test 2

Is this correct?  I've build my own parser for quill delta which works well except for alignment.  How could you logically tackle having the `align:center` attribute when no where is it specified that Test 1 and Test 2 would have a different attribute?

Thanks!

**Comment by aldycool** [comment]
I think that's because alignment is a Block Style, not a Line Style. Block Style is applied at the end of the block itself. Actually your "Test 1" line is also contained in a block, but not specifying its alignment, which means defaulted to Left Align.

**Reply by mark8044** [reply]
Ok I get it now.. blocks!

Thanks!!
